libimage/filters.go

```diff
@@ -159,8 +159,9 @@ func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOp
 			filter = filterIntermediate(ctx, intermediate, t)
 
 		case "label":
-			filter = filterLabel(ctx, value)
-
+			filter = filterLabel(ctx, value, false)
+		case "label!":
```

Comment:
Can you extend the tests in fitlers_test.go?

Reply:
Looking at line `101` I don't think this is the right thing to do here.  Can you investigate?